### PR TITLE
Use config instead of hard coded teams.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/25/

File: idp/usso/usso.go [hunk 1]
```diff
@@ -34,42 +34,54 @@ import (
...
-	groupCache:     cache.New(10 * time.Minute),
-	groupMonitor: prometheus.NewSummary(prometheus.SummaryOpts{
-		Namespace: "candid",
-		Subsystem: "launchpad",
-		Name:      "get_launchpad_groups",
-		Help:      "The duration of launchpad login, /people, and super_teams_collection_link requests.",
-	}),
+type Params struct {
+	// openIdRequestedTeams contains any private teams that the system needs to know about.
+	OpenIdRequestedTeams []string `yaml:"openId-requested-teams"`
```
Comment (rogpeppe) [inline idp/usso/usso.go]: Not convinced about the combination of camel and snake case here.
How about

    PrivateTeams []string `yaml:"private-teams"`

?

File: idp/usso/usso.go [hunk 2]
```diff
@@ -34,42 +34,54 @@ import (
...
+}
+
+// This list needs to contain default private teams that the system needs to know about.
+var defaultOpenIdRequestedTeams = []string{"blues-development", "charm-beta"}
```
Comment (rogpeppe) [inline idp/usso/usso.go]: I'm not sure this should be the default any more. Given that we're changing the charm, I think we can just default to empty and change the charm config at upgrade time.

File: cmd/candidsrv/main.go [hunk 3]
```diff
@@ -213,6 +213,6 @@ func serveIdentity(conf *config.Config, params identity.ServerParams) error {
 }
 
 var defaultIDPs = []idp.IdentityProvider{
-	usso.IdentityProvider,
+	usso.NewIdentityProvider(usso.Params{}),
 	ussooauth.IdentityProvider,
```
Comment (mhilton) [inline cmd/candidsrv/main.go]: Take the oauth version out of defaults (as it's easy to do here).

File: idp/usso/usso.go (hunk 1 above)
Comment (mhilton) [reply idp/usso/usso.go]: This isn't part of the openid protocol itself, but a launchpad extension so why not "launchpad-teams"?

File: idp/usso/usso.go (hunk 2 above)
Comment (mhilton) [reply idp/usso/usso.go]: +1

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/28/

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**